pkg/services/alerting/service.go

```diff
@@ -116,6 +119,24 @@ func (s *AlertNotificationService) GetAlertNotificationsWithUid(ctx context.Cont
 }
 
 func (s *AlertNotificationService) UpdateAlertNotificationWithUid(ctx context.Context, cmd *models.UpdateAlertNotificationWithUidCommand) error {
+	var err error
+	cmd.EncryptedSecureSettings, err = s.EncryptionService.EncryptJsonData(ctx, cmd.SecureSettings, setting.SecretKey)
+	if err != nil {
+		return err
+	}
+
+	model := models.AlertNotification{
+		Uid:      cmd.NewUid,
+		OrgId:    cmd.OrgId,
+		Name:     cmd.Name,
+		Type:     cmd.Type,
+		Settings: cmd.Settings,
+	}
+
+	if err := s.validateAlertNotification(ctx, &model, cmd.SecureSettings); err != nil {
+		return err
+	}
+
 	return s.SQLStore.UpdateAlertNotificationWithUid(ctx, cmd)
```

Comment:
Nit. Using this one feels a bit weird now when we have a service since this calls the `sqlstore.UpdateAlertNotification(ctx, cmd). More optimal if service method would look up the alert notification by UID using `sqlstore.GetAlertNotificationsWithUid` and mimic the [sqlstore.UpdateAlertNotificationWithUid](https://github.com/grafana/grafana/blob/c6943797f94af7f793e9d4f7dad533897c6a1ef3/pkg/services/sqlstore/alert_notification.go#L452-L483) - then I think we can remove the `sqlstore.UpdateAlertNotificationWithUid(...)` since no longer in use. 

Think we need to fix this because `createNotifier` checks `if model.Id > 0 {` which will never be set when updating using `UID` so there's a chance already stored secure settings being overwritten.